src/Sylius/Bundle/SettingsBundle/Model/SettingsInterface.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\SettingsBundle\Model;
+
+use Doctrine\Common\Collections\Collection;
+use Sylius\Component\Resource\Model\ResourceInterface;
+
+/**
+ * @author Steffen Brem <steffenbrem@gmail.com>
+ */
+interface SettingsInterface extends ResourceInterface
+{
+    /**
+     * @return string
+     */
+    public function getSchemaAlias();
+
+    /**
+     * @param string $schemaAlias
+     */
+    public function setSchemaAlias($schemaAlias);
+
+    /**
+     * @return string
+     */
+    public function getNamespace();
+
+    /**
+     * @param string $namespace
+     */
+    public function setNamespace($namespace);
+
+    /**
+     * @return Collection|ParameterInterface[]
+     */
+    public function getParameters();
+
+    /**
+     * @param string $name
+     *
+     * @throws \InvalidArgumentException
```

Comment:
It's the way it communicates. You can either do `@return string|null` or `@return string @throws \FooBarException`, the second one is less error prone. 

It's good, but I'll vote for a custom exception (`ParameterNotFoundException`?).